benches/benches/vm.rs

```diff
@@ -38,8 +38,11 @@ where
             let mut total = core::time::Duration::ZERO;
             for _ in 0..iters {
                 let original_db = vm.as_mut().database_mut().clone();
-                let database_tx = original_db.transaction().as_ref().clone();
-                *vm.as_mut().database_mut() = database_tx;
+                // Simulates the block production/validation with three levels of database transaction.
```

Comment:
I'm not following this what this is for. These each seem to be converted to the same `DatabaseTransaction` type under the hood.

Reply:
It is converted to the same type, but inclusion depth increases with each `DatabaseTransaction`. 

During the execution of the block we have several layers of `DatabaseTransaction`.

When you call `get` on `DatabaseTransaction`, it goes to the internal database, which is `DatabaseTransaction`, that goes to another `DatabaseTransaction` that goes to `RocksDB`. 3 layers of `DatabaseTransaction` simulates the real execution environment complexity.

Reply:
Hmm. Okay. This might be naive of me, but I feel like we should be benching against the surface _above_ where this happens, that way we don't have to simulate it. That might not be possible, and I'm not able to suggest a way to do it, so I'm not gonna push too hard on that.